src/net/sourceforge/kolmafia/textui/parsetree/BasicScope.java

```diff
@@ -122,84 +122,47 @@ public final Function findFunction(
     }
 
     // We will consider functions from this scope and from the RuntimeLibrary.
-    Function[] userFunctions = this.functions.findFunctions(name);
-    Function[] libraryFunctions = RuntimeLibrary.functions.findFunctions(name);
+    Function[] functions = this.functions.findFunctions(name);
```

Comment:
One other possibility that's used by other languages like Python is to have a separate scope for builtins so if people _really_ want to override builtin functions, they can.

e.g. builtin scope -> global scope -> function-level scope

Reply:
well, isn't this already how ASH is doing it?

Reply:
Not quite. You can shadow a user-defined function at a different scope, but you cannot shadow a builtin function no matter what you do.

For instance:
```
string a() {
	return "global";
}

print(a());

void main() {
  string a() {
    return "nested";
  }
  print(a());
}
```
will print:
```
global
nested
```